### PR TITLE
Rename all titled Java*

### PR DESCRIPTION
**Pull request by lanking520**
Rename all of the APIs

@andrewfayres

### CODE REVIEW

File: scala-package/core/src/main/scala/org/apache/mxnet/javaapi/IO.scala [hunk 1]
```diff
@@ -15,21 +15,18 @@
  * limitations under the License.
  */
 
-package org.apache.mxnet.api.java
+package org.apache.mxnet.javaapi
```
Comment (piyushghai) [inline scala-package/core/src/main/scala/org/apache/mxnet/javaapi/IO.scala:18]: java.api

**Comment by piyushghai** [comment]
Nitpick, instead of a folder named javaapi, let's make it java/api so that package name becomes java.api

File: scala-package/core/src/main/scala/org/apache/mxnet/javaapi/NDArray.scala [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+  implicit def fromNDArray(nd: org.apache.mxnet.NDArray): NDArray = new NDArray(nd)
+
+  implicit def toNDArray(jnd: NDArray): org.apache.mxnet.NDArray = jnd.nd
+
+  def waitall(): Unit = org.apache.mxnet.NDArray.waitall()
+
+  def onehotEncode(indices: NDArray, out: NDArray): NDArray = org.apache.mxnet.NDArray.onehotEncode(indices, out)
+
+  def empty(shape: Shape, ctx: Context, dtype: DType.DType): NDArray
+  = org.apache.mxnet.NDArray .empty(shape, ctx, dtype)
```
Comment (andrewfayres) [inline scala-package/core/src/main/scala/org/apache/mxnet/javaapi/NDArray.scala]: Minor nitpick. A few of these have spaces between the NDArray and empty.

**Review by andrewfayres — APPROVED** [review]

**Comment by andrewfayres** [comment]
LGTM. We'll talk about the package name and try to find something better than javaapi but this is fine for now.

**Pull request merged**